util/log/context.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tobias Schottdorf
+
+package log
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/net/context"
+)
+
+// Context wraps a context.Context into an object suitable for logging.
+type Context interface {
+	context.Context
+	With(kvs ...interface{}) Context
+
+	Infof(string, ...interface{})
```

Comment:
I'd rather keep these as static functions in the log package so we don't have to change the interface and can continue to use context.Context as the type that gets passed around everywhere.